yt/yt/ytlib/api/native/transaction.cpp

```diff
@@ -352,17 +353,21 @@ class TTransaction
         i64 newOffset,
         const TAdvanceConsumerOptions& /*options*/) override
     {
-        auto tableMountCache = GetClient()->GetTableMountCache();
-        auto queuePhysicalPath = queuePath;
-        auto queueTableInfoOrError = WaitFor(tableMountCache->GetTableInfo(queuePath.GetPath()));
-        if (!queueTableInfoOrError.IsOK()) {
-            THROW_ERROR_EXCEPTION("Failed to resolve queue path")
-                << queueTableInfoOrError;
+        CheckConsumerPermission(Client_->GetNativeConnection(), consumerPath, Client_->GetOptions());
+
+        auto registrationCheckResult = Client_->GetNativeConnection()->GetQueueConsumerRegistrationManager()->GetRegistration(queuePath, consumerPath);
+        if (!registrationCheckResult.Registration) {
```

Comment:
Checking registrations should also go inside the helper, IMO, since it is also exactly the same.